pkgs/development/compilers/cudatoolkit/redist/build-cuda-redist-package.nix

```diff
@@ -1,66 +1,175 @@
-{ lib
-, stdenv
-, backendStdenv
-, fetchurl
-, autoPatchelfHook
-, autoAddOpenGLRunpathHook
+# Type Aliases
+#
+# See ./extension.nix:
+# - ReleaseAttrs
+# - ReleaseFeaturesAttrs
+#
+# General callPackage-supplied arguments
+{
+  autoAddOpenGLRunpathHook,
+  autoPatchelfHook,
+  backendStdenv,
+  fetchurl,
+  lib,
+  lndir,
+  pkgsTargetTarget,
```

Comment:
I thought this should be `pkgsHostTarget` (=plain `pkgs`), btw.

E.g. you're on a x86_64 device and you want to produce a library (e.g. libstdc++) to be run an aarch64 host:

```
❯ # Where the derivation is built
❯ nix eval nixpkgs#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.buildPlatform.system
"x86_64-linux"
❯ # Where the final executable is to be run
❯ nix eval nixpkgs#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.hostPlatform.system
"aarch64-linux"
```

`nixpkgs#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.pkgsTargetTarget` looks empty to me o_0

Reply:
@samuela @SomeoneSerge 

I had found that `pkgsTargetTarget` worked while `pkgsHostTarget` did not.

I believe this is because `pkgsHostTarget` has packages which are built with a compiler which runs on `Host` (first component) and produces binaries for `Target` (second component).

Since we want to ensure we link against the `libc` that the target will be using, we need to grab a copy of the compiler/standard library which is meant to run on `Target`. That's why I use `pkgsTargetTarget` -- so the compiler/standard library we grab is meant to run on `Target` (first component).

[Using the section on bootstrapping as a reference](https://nixos.org/manual/nixpkgs/stable/#ssec-bootstrapping), by using `pkgsTargetTarget` we're effectively grabbing the compiler/standard library from the last stage.

Does that make sense? I may be interpreting this incorrectly.

Reply:
I see that llvm appears to be using `pkgsTargetTarget` as well:

https://github.com/NixOS/nixpkgs/blob/c58bd8261fde336b04fb69c57171d33c7342f8a5/pkgs/top-level/all-packages.nix#L16343-L16344


But no, it does not make sense to me at this point, nor does the documentation 😩

Reply:
Another thing adding to complexity: `pkgsTargetTarget` is an edge forward, but `stdenv` is an edge back, because `stdenv.cc`'s host platform is always `buildPlatform`. E.g. in `nixpkgs#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform`, `gcc`'s host is aarch64, but `stdenv.cc`'s host is x86_64.

1. Can't we actually drop both `stdenv` and `pkgsTargetTarget`? Isn't it the case with splicing, that if we put `cc.lib` into the correct `deps<host><target>` (`buildInputs`, in our case?), then we don't have to access `pkgs<host><target>` manually?
2. Either way, you can make the decision in the cross-compilation PR
3. Would it perhaps make sense to move the libstdc++ parameter into `cudaPackages` attributes and accept it as an .override-able argument like so: https://github.com/SomeoneSerge/nixpkgs/pull/5/files#diff-39a8f9c15e23296b8362fc5df0027cb4439e026a8c3793eaf6179353578ffb29R74-R76 . This can be convenient for debugging, e.g. you could do `nix build .#cudaPackages.gccForLibs.lib ; readelf -h result-lib/aarch64-unknown-linux-gnu/lib/libstdc++.so`, `nix build .#buildPackages.cudaPackages.gccForLibs.lib`, etc

Reply:
Also, denoting `nixpkgs#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform` by `pkgs`, I think `pkgs.pkgsTargetTarget` just isn't a thing, it's only `pkgs.buildPackages.pkgsTargetTarget` that makes sense

Reply:
@SomeoneSerge those comments are gold!

I've decided to defer the thing until multi-arch support to try to get this PR back on track and ready to merge.